### DISCUSSION TITLE
Recover signals after applying phase sequence

### DISCUSSION THREAD

**Discussion by duxiao1992** [NVlabs/sionna#501] [General]
Hello guys,

I am trying to reproduce the 'MIMO OFDM Transmissions over the CDL Channel Model' the the tutorial,  the configuration is exactly the same as tutorial.  Here is what I add base on the demo, for each OFDM symbol, I multiply a phase sequence (length of fft_size) before modulating.  Assuming both transmitter and receiver know the phase sequence of all OFDM symbols, since the tensor the transmitter sent has different shape with tensor the receiver got,  how can I recover to the original signal.

Thanks for help

**Comment by jhoydis** [comment]
Hi @duxiao1992,

It is difficult to see the exact issue without a code example, but I have the feeling that the problem might be that you apply the phase sequence to the output of the `OFDMResourceGridMapper` and not the `QAMMapper`. 
 The `OFDMResourceGridMapper` inserts pilot signals in the resource grid which are not considered as data.

The shape of the output of the `LMMSEEqualizer` should be identical to that of the `QAMMapper`. If you multiply the output of the `LMMSEEqualizer` by the complex conjugate of the phase sequence used to multiply the outputs of the `QAMMapper`, it should work.

Hope that points you to the right direction to solve your problem.
